README.md

```diff
@@ -187,6 +187,21 @@ Note, in Postgres streaming is going to be slower than non-streaming options due
 
 Streaming support is only implemented in the postgres backend at the moment, PRs welcome to add to other backends.
 
+## Prepared Statements
+See [benchmark mini_sql](https://github.com/discourse/mini_sql/tree/master/bench/prepared_perf.rb)
+[benchmark mini_sql vs rails](https://github.com/discourse/mini_sql/tree/master/bench/bilder_perf.rb).
+
+By default prepared cache size is 500 queries. Use prepared queries only for frequent queries.
+
+```ruby
+conn.prepared.query("select * from table where id = ?", id: 10)
+
+ids = rand(100) < 90 ? [1] : [1, 2]
+builder = conn.build("select * from table /*where*/")
+builder.where("id IN (?)", ids)
+builder.prepared(ids.size == 1).query # most frequent query
+```
```

Comment:
oh I just realized, how do we clean up and "unprepare" a query?

```
conn.prepared.clear_prepared_query("select * from table where id = ?")
```
Since we have multi param stuff that is dynamic ... maybe we only support

```
conn.prepared.clear_cache!
```

Reply:
Can you give an example of when this method is needed in a real application?

now only [500 prepared queries are cached](https://github.com/discourse/mini_sql/pull/20/files#diff-0d800dee43437a31a7ea8beece30961c82749269c6739e6d34b95896b44b7e54R23), if the limit is exceeded, the last unused prepared query[ will be deallocated](https://github.com/discourse/mini_sql/pull/20/files#diff-61dbd369892c0d66cedd249a9a95be54821ae82e94f287b9fe2cb1ae5fa3c490R50)

also, when the web worker (puma unicorn) killing, the database connections are automatically closed, and the prepared queries on current connection automatically deleted 

https://www.postgresql.org/docs/10/sql-prepare.html

> Prepared statements only last for the duration of the current database session. When the session ends, the prepared statement is forgotten, so it must be recreated before being used again.

Reply:
I follow, was thinking of future proofing here in case users opt to clean up early for unknown reasons. In practice I guess I am not expecting this to happen. Let's merge, adding a clear later is trivial anyway.